resources/rollup-config.mjs

```diff
@@ -33,15 +40,24 @@ export default [
         plugins: [terser()],
       },
       // es build for bundlers and node
-      {
-        banner: copyright,
-        name: 'Immutable',
-        dir: path.join(DIST_DIR, 'es'),
-        format: 'es',
-        preserveModules: true,
-        preserveModulesRoot: SRC_DIR,
-        sourcemap: false,
-      },
+      CIRCULAR_DEPS_IN_SRC_DIR
+        ? {
+            banner: copyright,
+            name: 'Immutable',
+            // generates one esm file with no circular imports
+            file: path.join(DIST_DIR, 'es', 'Immutable.js'),
```

Comment:
As a matter of fact, it was like that before trying to resolve tree-shaking in https://github.com/immutable-js/immutable-js/commit/8dffce49aedea700706a968895e1d0f19ef0a1ed#diff-a098417b5f500a442eab89048eb0705bf6acad8b8f76514b70d27ba2e10f55a5

But we can not resolve tree-shaking [because everything is connected](https://github.com/immutable-js/immutable-js/issues/1961#issuecomment-1727029822), so let's rollback to a single file 👍